### PR TITLE
add opt-in statement to newsletter subscription

### PR DESCRIPTION
**Pull request by alex-mo**
- provide a field on the component to set a statement for internal
  audit trail (will not be shown to users)
- save the text along the newsletter_subscription activity
- the activity is saved on confirmation (not on creation)

### CODE REVIEW

**Review by torotil — CHANGES_REQUESTED** [review]
Looks good in general. Could you try to also meet these criteria?

* Fix coding-style issues in all non-features files touched by the change (in a separate commit).
* Add tests (for the changes) so that the coverage over-all coverage doesn’t decrease.

I know this sets the bar higher for new code than it was for the old, but then again that’s the point isn’t it?

File: campaignion_newsletters/src/Subscription.php [hunk 1]
```diff
@@ -23,6 +28,14 @@ class Subscription extends \Drupal\little_helpers\DB\Model {
   protected static $values = array('fingerprint', 'updated', 'last_sync');
   protected static $serial = FALSE;
 
+  /**
+   * Construct a new instance from data found in the DB.
```
Comment (torotil) [inline campaignion_newsletters/src/Subscription.php]: Mh that should be more like "Get an instance form DB or create a new one." or something like that.

File: campaignion_newsletters/src/Subscription.php [hunk 2]
```diff
@@ -31,18 +44,36 @@ public static function byData($list_id, $email) {
...
+   */
   public static function fromData($list_id, $email) {
     return new static(array(
       'list_id' => $list_id,
       'email' => $email,
     ), TRUE);
   }
 
+  /**
+   * Construct a new instance from email addresses found in the DB.
```
Comment (torotil) [inline campaignion_newsletters/src/Subscription.php]: That should be "Load all subscriptions for one email address." or something similar.

File: campaignion_newsletters/src/Subscription.php [hunk 3]
```diff
@@ -55,6 +86,9 @@ public static function byEmail($email) {
     return $subscriptions;
   }
 
+  /**
+   * Get the newsletter list instance.
```
Comment (torotil) [inline campaignion_newsletters/src/Subscription.php:90]: This needs a @return docs.

**Review by torotil — COMMENTED** [review]
Some of the method summaries are still a bit imprecise, but it definitely looks much better now.

File: campaignion_newsletters/src/Subscription.php (hunk 1 above)
Comment (alex-mo) [reply campaignion_newsletters/src/Subscription.php]: fine with me

File: campaignion_newsletters/src/Subscription.php (hunk 3 above)
Comment (alex-mo) [reply campaignion_newsletters/src/Subscription.php:90]: `phpcs` is silent about that...

**Review by torotil — APPROVED** [review]

**Comment by torotil** [comment]
Great, thanks!

**Pull request merged**